src/window/city.c

```diff
@@ -498,6 +498,18 @@ static void handle_hotkeys(const hotkeys *h)
         game_undo_perform();
         window_invalidate();
     }
+    if (h->mothball_toggle) {
+        int building_id = map_building_at(widget_city_current_grid_offset());
+        if (building_id) {
+            building *b = building_main(building_get(building_id));
+            building_mothball_toggle(b);
+            if (b->state == BUILDING_STATE_IN_USE) {
+                city_warning_show(WARNING_DATA_MOTHBALL_OFF, NEW_WARNING_SLOT);
```

Comment:
I think you can store the warning id as a static variable for this file and use it like:
```
data.mothball_warning_slot = city_warning_show(WARNING_DATA_MOTHBALL_OFF, data.mothball_warning_slot);
```

And do the same two lines below. 

That way it will reuse the same line when mothballing/unmothballing, instead of showing both at the same time, confusing the player.

Reply:
When I just change:
 `city_warning_show(WARNING_DATA_MOTHBALL_OFF, NEW_WARNING_SLOT);`

to 

`city_warning_show(WARNING_DATA_MOTHBALL_OFF, 1);`

the same ist done?!

Reply:
Am I wrong or could I just define a new warning slot?!

city/warning.h
`#define MOTHBALL_WARNING_SLOT 1`

city.c
`city_warning_show(WARNING_DATA_MOTHBALL_OFF, MOTHBALL_WARNING_SLOT);`

this way its also shown once.

P.S.: Im more of a modder than a coder, I have just a basic understanding of the code, so I would appreciate if you could go more into detail on your static var proposal.
Could you post an example how you had done it, so I can learn from it? I think augustus is a nice project to get/refresh my c-skills, which are very rusty. (Because of my limited understanding I have a hardtiume understanding this data thing, which seems very handy and is used often, but I could not "decipher" it LOL

Reply:
OK in the meanwhile I also added an hotkey to show the storage_order by one click, I would push this as soon as I got the static var thing for the warning_window, a further explanation will be highly appreciated (I know this is a basic thing, so I need the explanation even more ;)

PS: I also would like to merge building info and storage info, to one window, but I assume with my skill, this will took a while :D

Reply:
I would say this would be the most simple (for me) solution:

`    if (h->mothball_toggle) {

        int building_id = map_building_at(widget_city_current_grid_offset());
        if (building_id) {
            building *b = building_main(building_get(building_id));
            building_mothball_toggle(b);
            if (b->state == BUILDING_STATE_IN_USE) {
               city_warning_clear_all();
                city_warning_show(WARNING_DATA_MOTHBALL_OFF, NEW_WARNING_SLOT);
            } else if (b->state == BUILDING_STATE_MOTHBALLED) {
                city_warning_clear_all();
                city_warning_show(WARNING_DATA_MOTHBALL_ON, NEW_WARNING_SLOT);
            }
        }
    }`

Reply:
I would say, this is the most simple (for me) solution:

`    if (h->mothball_toggle) {

        int building_id = map_building_at(widget_city_current_grid_offset());
        if (building_id) {
            building *b = building_main(building_get(building_id));
            building_mothball_toggle(b);
            if (b->state == BUILDING_STATE_IN_USE) {
               city_warning_clear_all();
                city_warning_show(WARNING_DATA_MOTHBALL_OFF, NEW_WARNING_SLOT);
            } else if (b->state == BUILDING_STATE_MOTHBALLED) {
                city_warning_clear_all();
                city_warning_show(WARNING_DATA_MOTHBALL_ON, NEW_WARNING_SLOT);
            }
        }
    }`

Reply:
I would say, this is the most simple solution (for me):


    if (h->mothball_toggle) {

        int building_id = map_building_at(widget_city_current_grid_offset());
        if (building_id) {
            building *b = building_main(building_get(building_id));
            building_mothball_toggle(b);
            if (b->state == BUILDING_STATE_IN_USE) {
                city_warning_clear_all();
                city_warning_show(WARNING_DATA_MOTHBALL_OFF, NEW_WARNING_SLOT);
            } else if (b->state == BUILDING_STATE_MOTHBALLED) {
                city_warning_clear_all();
                city_warning_show(WARNING_DATA_MOTHBALL_ON, NEW_WARNING_SLOT);
            }
        }
    }

Reply:
The reason keeping the id won't work is due to the way the warning system is coded.

An id is only valid *while the corresponding warning slot is being shown*. As soon as that slot disappears, the id will be permanently invalid, even if a new warning goes to that very same slot.

So when sending an id with the function, you say,  "if the provided id is still valid, overwrite the warning slot with this new message".

That's why the function also returns an id. It may still be the same id (if the message was still active and showing) or a new one, in which case you should replace the old id with the new returned one.

The id `NEW_WARNING_SLOT` is simply an id of `0`, which id always inactive so it will create a new slot every time (and a new id which you can ignore).

Reply:
Also, your proposed code isn't entirely correct as it will also remove other messages that may be showing that are unrelated to the mothball status of a building.

Reply:
Hello, thank you for the response.
While this:
 void city_warning_clear_all(void)
{
    for (int i = 0; i < MAX_WARNINGS; i++) {
        warnings[i].in_use = 0;
    }
}
is very clear to me, aswell as most parts of warning.c , I was still not able to get the correct solution (Sunday).
I appreciate your effort to show me the right direction, I will continue my "investigation" next weekend.
Have a nice day :)

Reply:
Sure thing, have fun!

Reply:
I still dont got it, for me the clear all solution is fine. 

"Also, your proposed code isn't entirely correct as it will also remove other messages that may be showing that are unrelated to the mothball status of a building."

Thats very true, but the same happens as as soon as you hit the advisor button in the topbar (to be exact: in every instance the city window is not active). So for me I will claim this effect "minor". I played the game since the change a very while and could not experience an instance when it was important to have the warnings stay. So for me its fine because I cant find a solution in a reasonable time. :/

Reply:
Yeah it's a minor issue indeed!

Can you just remove the encoder English strings from the non English languages please?